src/nestor/handlers/contacts.py

```diff
@@ -143,9 +148,49 @@ def __add_contact(self) -> str:
             message = Colorizer.success(f"Contact {name} updated.")
 
         return message
+    
+    @input_error({ValueError: "Contact editing interrupted. Contact not updated.", IndexError: "Name od the contact is required"})
+    def __edit_contact(self, *args) -> str:
+        name = args[0]
+
+        contact = self.book.find(name)
+
+        if contact is None:
+            message = Colorizer.warn("Contact not found.")
+
+        """
+        Edits contact in contacts dictionary
+        """
+
+        fields = [
+            FieldInput(prompt=f"Name ({contact.name})", validator=Name.validate, is_required=False),
+            FieldInput(prompt=f"Phone number ({'; '.join([str(item) for item in contact.phones]) if len(contact.phones) else 'None'})", validator=Phone.validate, is_required=False),
+            FieldInput(prompt=f"Date of Birth ({contact.birthday})", validator=Birthday.validate, is_required=False),
+            FieldInput(prompt=f"Email ({contact.email})", validator=Email.validate, is_required=False),
+        ]
+
+        new_name, phone, date, email = command_data_collector(fields)
+
+        if new_name and new_name != contact.name and self.book.find(new_name):
+            return Colorizer.warn(f"Contact with name '{new_name}' already exist.")
+        elif new_name and new_name != contact.name:
+            contact = copy.deepcopy(contact)
```

Comment:
why we are coping, then delete, add, and then update?

Reply:
Because contacts are stored in the book dictionary by key=name, so we need to remove a contact stored by the old name key and add it back by the new name key.